Best practices/Design patterns/Behavioral patterns.md

```diff
@@ -1,3 +1,47 @@
+#### Strategy pattern
+
+By using this pattern we enable a selection of needed algorithms at runtime. For instance, validation is dependent on the incoming type of data. We don't know beforehand which algorithm will be needed, but we can develop a few strategies and use them when needed. Algorithms should be defined in a way that they could be used interchangeably.
```

Comment:
I think that this explanation could be improved a bit. Although it is technically correct, if I never heard of the pattern before and only read this description, I wouldn't be exactly sure in what cases should this be used and how. The code example below explains quite a lot, but nevertheless, I think that we could do a better job here too.

Reply:
@icottiero I agree. The explanation has been improved with additional information to clarify when and why we would use the algorithm.